src/tainting/Taint.ml

```diff
@@ -105,18 +120,24 @@ let _show_finding = function
 type orig = Src of source | Arg of arg_pos [@@deriving show]
 type taint = { orig : orig; tokens : tainted_tokens } [@@deriving show]
 
-let src_of_pm (pm, x) = Src (PM (pm, x))
+let src_of_pm (pm, (x : Rule.taint_source)) =
+  Src { call_trace = PM (pm, x); label = x.label }
+
 let taint_of_pm pm = { orig = src_of_pm pm; tokens = [] }
 
 let compare_sources s1 s2 =
   (* Comparing metavariable environments this way is not robust, e.g.:
    * [("$A",e1);("$B",e2)] is not considered equal to [("$B",e2);("$A",e1)].
    * For our purposes, this is OK.
    *)
-  let pm1, ts1 = pm_of_trace s1 and pm2, ts2 = pm_of_trace s2 in
-  Stdlib.compare
-    (pm1.rule_id, pm1.range_loc, pm1.env, ts1.Rule.label)
-    (pm2.rule_id, pm2.range_loc, pm2.env, ts2.Rule.label)
+  let pm1, ts1 = pm_of_trace s1.call_trace
+  and pm2, ts2 = pm_of_trace s2.call_trace in
+  match String.compare s1.label s2.label with
+  | 0 ->
+      Stdlib.compare
+        (pm1.rule_id, pm1.range_loc, pm1.env, ts1.Rule.label)
```

Comment:
Why not just
```ocaml
   Stdlib.compare
     (pm1.rule_id, pm1.range_loc, pm1.env, s1.label, ts1.Rule.label)
     (pm2.rule_id, pm2.range_loc, pm2.env, s2.label, ts2.Rule.label)
```

Reply:
ooh, good point. i forgot polymorphic compare was, well, polymorphic